app/components/elements/Author.jsx

```diff
@@ -11,32 +11,56 @@ import Userpic from 'app/components/elements/Userpic';
 import { translate } from 'app/Translator';
 import normalizeProfile from 'app/utils/NormalizeProfile';
 
-const {string, bool, number} = React.PropTypes
+const {string, bool, number} = React.PropTypes;
 
 class Author extends React.Component {
     static propTypes = {
         author: string.isRequired,
         follow: bool,
         mute: bool,
         authorRepLog10: number,
-    }
+    };
     static defaultProps = {
         follow: true,
         mute: true,
+    };
+
+    constructor() {
+        super();
+        this.showProfileCtrl = this.showProfileCtrl.bind(this);
+    }
+
+    componentDidMount() {
+        const element = document.querySelector('.FoundationDropdownMenu__label');
+        if (element) element.addEventListener("click", this.showProfileCtrl, false);
+    }
+
+    componentWillUnmount() {
+        const element = document.querySelector('.FoundationDropdownMenu__label');
+        if (element) element.removeEventListener("click", this.showProfileCtrl, false);
     }
-    shouldComponentUpdate = shouldComponentUpdate(this, 'Author')
+
+    showProfileCtrl(e) {
+        if (e.metaKey || e.ctrlKey) { // handle edge case for ctrl clicks
```

Comment:
is this the right way to be doing this?  control-clicks and middle-clicks should probably be handled via the same code path versus special-casing them?